NewForce-Cohort-5/tabloidmvc-the-daily-debuggers
# User posts np

# Description

I created a MyPosts view that allows users to see their posts sorted by creation date. Users can VIEW, EDIT, and DELETE their posts from MyPosts view only. Users can no longer EDIT or DELETE posts from Posts view.


#Type of changes

[Check Trello Ticket #4](https://trello.com/b/ECtaPfZr/the-daily-debuggers)



# Testing Instructions

1. git add and git commit all changes in your current branch.
2. git checkout main
3. git fetch -a
4. git checkout user-posts-np
5. Navigate to the root directory for this repo (directory-to)/tabloidmvc-the-daily-debuggers
6. Start TabloidMVC.sln
7. Make sure your db is created with seed data (Skip to 9 if this is already done)
8. Run [01_TabloidMVC_Create_DB.sql](https://github.com/NewForce-Cohort-5/tabloidmvc-the-daily-debuggers/blob/main/SQL/01_TabloidMVC_Create_DB.sql) and [02_TabloidMVC_Seed_Data.sql](https://github.com/NewForce-Cohort-5/tabloidmvc-the-daily-debuggers/blob/main/SQL/02_TabloidMVC_Seed_Data.sql) queries in Visual Studio
9. Start the server in Debug Mode F5
10. Log in as admin admin@example.com (if not already logged in as admin)
11. Navigate to the posts link
12. Navigate to my posts
13. Make sure you see only your posts and have the ability to VIEW, EDIT, and DELETE

Use the following SQL query to see what you should expect to see on the website
```sql
SELECT p.Id, p.Title, p.Content, 
                              p.ImageLocation AS HeaderImage,
                              p.CreateDateTime, p.PublishDateTime, p.IsApproved,
                              p.CategoryId, p.UserProfileId,
                              c.[Name] AS CategoryName,
                              u.FirstName, u.LastName, u.DisplayName, 
                              u.Email, u.CreateDateTime, u.ImageLocation AS AvatarImage,
                              u.UserTypeId, 
                              ut.[Name] AS UserTypeName
                         FROM Post p
                              LEFT JOIN Category c ON p.CategoryId = c.id
                              LEFT JOIN UserProfile u ON p.UserProfileId = u.id
                              LEFT JOIN UserType ut ON u.UserTypeId = ut.id
                        WHERE IsApproved = 1 AND PublishDateTime < SYSDATETIME()
```
# Checklist:
